services/provider/server/server.go

```diff
@@ -925,6 +932,37 @@ func extractMonitorIps(data string) ([]string, error) {
 	return ips, nil
 }
 
-func (s *OCSProviderServer) PeerStorageCluster(_ context.Context, _ *pb.PeerStorageClusterRequest) (*pb.PeerStorageClusterResponse, error) {
+func (s *OCSProviderServer) PeerStorageCluster(ctx context.Context, req *pb.PeerStorageClusterRequest) (*pb.PeerStorageClusterResponse, error) {
+
+	pubKey, err := s.getOnboardingValidationKey(ctx)
+	if err != nil {
+		klog.Errorf("failed to get public key to validate peer onboarding ticket %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to validate peer onboarding ticket")
+	}
+
+	onboardingToken, err := decodeAndValidateTicket(req.OnboardingToken, pubKey)
+	if err != nil {
+		klog.Errorf("Invalid onboarding token. %v", err)
+		return nil, status.Errorf(codes.InvalidArgument, "onboarding ticket is invalid")
```

Comment:
Please use consistent phrasing
```suggestion
		return nil, status.Errorf(codes.InvalidArgument, "invalid onboarding ticket")
```